### PR TITLE
Pods 9711 - Twirl migration code changes 

### PR DESCRIPTION
**Pull request by jamdade-sharad**

### CODE REVIEW

**Comment by rambabu-posa** [comment]
I think, most of the specs are using application.
This should be override def fakeApplication: Application instead of private val application: Application

File: app/services/common/details/CommonDateOfBirthService.scala [hunk 1]
```diff
@@ -42,13 +43,12 @@ import scala.util.Try
 
 @Singleton
 class CommonDateOfBirthService @Inject()(val controllerComponents: MessagesControllerComponents,
-                                         val renderer: Renderer,
+                                         dobView: DobView,
                                          val userAnswersCacheConnector: UserAnswersCacheConnector,
                                          val navigator: CompoundNavigator,
                                          val messagesApi: MessagesApi)
   extends NunjucksSupport with FrontendHeaderCarrierProvider with I18nSupport with Retrievals {
```
Comment (rambabu-posa) [inline app/services/common/details/CommonDateOfBirthService.scala:50]: Please remove NunjucksSupport here

File: app/services/common/details/CommonDateOfBirthService.scala [hunk 2]
```diff
@@ -27,13 +27,14 @@ import play.api.data.FormBinding.Implicits._
 import play.api.i18n.{I18nSupport, MessagesApi}
 import play.api.libs.json.{Json, OWrites}
 import play.api.mvc.Results.{BadRequest, Ok, Redirect}
-import play.api.mvc.{AnyContent, MessagesControllerComponents, Result}
+import play.api.mvc.{AnyContent, Call, MessagesControllerComponents, Result}
 import renderer.Renderer
```
Comment (rambabu-posa) [inline app/services/common/details/CommonDateOfBirthService.scala:31]: Remove Renderer import here

File: app/views/DeclarationView.scala.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@this(
+    layout: templates.Layout,
+    formHelper: FormWithCSRF,
+    govukErrorSummary: GovukErrorSummary,
+    govukButton: GovukButton
+)
+
+@(schemeName: String, isCompany: Boolean, hasWorkingKnowledge: Boolean, submitCall: Call)(implicit request: Request[_], messages: Messages)
+
+@layout(messages("messages__declaration__title")) {
```
Comment (rambabu-posa) [inline app/views/DeclarationView.scala.html:26]: Do we need to add this param `formHasErrors = form.hasErrors` to layout?

Comment (jamdade-sharad) [reply app/views/DeclarationView.scala.html:26]: not needed in declarationView

File: test/base/SpecBase.scala [hunk 4]
```diff
@@ -87,11 +87,15 @@ trait SpecBase
                                       result: Future[Result],
                                       view: Html
                                     ): Assertion = {
-    org.scalatest.Assertions.assert(
+    println(s"*****")
+    println(s"${play.api.test.Helpers.contentAsString(result)(1.seconds).removeAllNonces().filterAndTrim}")
+    println(s"*****")
+    println(s"${view.toString().filterAndTrim}")
+    println(s"*****")
```
[the remark this answers is not in the document]
Comment (jamdade-sharad) [reply test/base/SpecBase.scala:94]: Removed

**Review by rambabu-posa — APPROVED** [review]
LGTM

**Pull request merged**